core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/valueCol.kt

```diff
@@ -0,0 +1,592 @@
+     * #### Examples for this overload:
+     *
+     * {@includeArg [CommonValueColDocs.ExampleArg]}
+     *
+     * To create a [ColumnAccessor] for another kind of column, take a look at the functions
+     * [col][ColumnsSelectionDsl.col],
+     * [colGroup][ColumnsSelectionDsl.colGroup],
+     * and [frameCol][ColumnsSelectionDsl.frameCol].
+     *
+     * @return A [ColumnAccessor] for the value column with the given argument if possible, else a [SingleColumn].
+     * @throws [IllegalStateException\] if the column with the given argument does not exist.
+     * @throws [IllegalArgumentException\] if the column with the given argument is not a value column.
+     *
+     * @see [valueColumn\]
+     * @see [ColumnsSelectionDsl.colGroup\]
+     * @see [ColumnsSelectionDsl.frameCol\]
+     * @see [ColumnsSelectionDsl.col\]
+     * {@arg [CommonValueColDocs.Note]}
+     */
+    private interface CommonValueColDocs {
+
+        /* Example argument, can be either {@include [SingleExample]} or {@include [DoubleExample]} */
+        interface ExampleArg
+
+        /**
+         * `df.`[select][DataFrame.select]` { {@includeArg [CommonValueColDocs.ReceiverArg]}`[valueCol][valueCol]`({@includeArg [CommonValueColDocs.Arg]}) \\\\}`
+         */
+        interface SingleExample
+
+        /**
+         * `df.`[select][DataFrame.select]` { {@includeArg [CommonValueColDocs.ReceiverArg]}`[valueCol][valueCol]`({@includeArg [CommonValueColDocs.Arg]}) \\\\}`
+         *
+         * `df.`[select][DataFrame.select]` { {@includeArg [CommonValueColDocs.ReceiverArg]}`[valueCol][valueCol]`<`[String][String]`>({@includeArg [CommonValueColDocs.Arg]}) \\\\}`
+         */
+        interface DoubleExample
+
+        /* Receiver argument for the example(s) */
+        interface ReceiverArg
+
+        /* Argument for the example(s) */
+        interface Arg
+
+        /* Optional note */
+        interface Note
+
+        /** @param [C\] The type of the value column. */
+        interface ValueColumnTypeParam
+    }
+
+    // region reference
+
+    /**
+     * @include [CommonValueColDocs]
+     * {@arg [CommonValueColDocs.Arg] valueColumnA}
+     * {@arg [CommonValueColDocs.ExampleArg] {@include [CommonValueColDocs.SingleExample]}}
+     * @param [col\] The [ColumnAccessor] pointing to the value column.
+     * @include [CommonValueColDocs.ValueColumnTypeParam]
+     */
+    private interface ValueColReferenceDocs
+
+    /**
+     * @include [ValueColReferenceDocs] {@arg [CommonValueColDocs.ReceiverArg]}
+     */
+    public fun <C> valueCol(valueCol: ColumnAccessor<C>): ColumnAccessor<C> = valueCol.ensureIsValueColumn()
+
+    /**
+     * @include [ValueColReferenceDocs] {@arg [CommonValueColDocs.ReceiverArg] myColumnGroup.}
+     */
+    public fun <C> SingleColumn<DataRow<*>>.valueCol(valueCol: ColumnAccessor<C>): SingleColumn<C> =
+        ensureIsColGroup()
+            .transformSingle {
+                it.getChild(valueCol)
+                    ?.cast<C>()
```

Comment:
Sometimes I really worry these paths of ???? because each of them multiplies the paths, do you think could it be handled otherwise?

Reply:
Not necessarily, in this specific case, only `getChild()` can return a nullable value for which the `IllegalStateException` will be thrown.
I continue the `?.` chain downwards for readability since something like 
```kotlin
(it.getChild(valueCol)?.cast<C>() ?: throw IllegalStateException("ValueColumn '${valueCol.path()}' not found in column group '${it.path}'"))
                    .also { it.data.ensureIsValueColumn() }
                    .let(::listOf)
```
becomes ugly and unreadable quickly. Other notation suggestions are welcome tho :)

Reply:
With AI help:
```kotlin
ensureIsColGroup().transformSingle {
    val child = it.getChild(valueCol)/*?.cast<IfNeeded>()*/
        ?: throw IllegalStateException("ValueColumn '${valueCol.path()}' not found in column group '${it.path}'")
    child.data.ensureIsValueColumn()
    listOf(child)
}.singleImpl()
```